src/plugins/inspector/public/views/requests/components/details/req_code_viewer.tsx

```diff
@@ -6,68 +6,111 @@
  * Side Public License, v 1.
  */
 
-import React from 'react';
+// Since we're not using `RedirectAppLinks`, we need to use `navigateToUrl` when
+// handling the click of the Open in Dev Tools link. We want to have both an
+// `onClick` handler and an `href` attribute so it will work on click without a
+// page reload, and on right-click to open in new tab.
+/* eslint-disable @elastic/eui/href-or-on-click */
+
+import { EuiButtonEmpty, EuiCopy, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { XJsonLang } from '@kbn/monaco';
-import { EuiFlexItem, EuiFlexGroup, EuiCopy, EuiButtonEmpty, EuiSpacer } from '@elastic/eui';
-
-import { CodeEditor } from '../../../../../../kibana_react/public';
+import { compressToEncodedURIComponent } from 'lz-string';
+import React, { MouseEvent, useCallback } from 'react';
+import { CodeEditor, useKibana } from '../../../../../../kibana_react/public';
 
 interface RequestCodeViewerProps {
+  indexPattern?: string;
   json: string;
 }
 
 const copyToClipboardLabel = i18n.translate('inspector.requests.copyToClipboardLabel', {
   defaultMessage: 'Copy to clipboard',
 });
 
+const openInDevToolsLabel = i18n.translate('inspector.requests.openInDevToolsLabel', {
+  defaultMessage: 'Open in Dev Tools',
+});
+
 /**
  * @internal
  */
-export const RequestCodeViewer = ({ json }: RequestCodeViewerProps) => (
-  <EuiFlexGroup
-    direction="column"
-    gutterSize="s"
-    wrap={false}
-    responsive={false}
-    className="insRequestCodeViewer"
-  >
-    <EuiFlexItem grow={false}>
-      <EuiSpacer size="s" />
-      <div className="eui-textRight">
-        <EuiCopy textToCopy={json}>
-          {(copy) => (
+export const RequestCodeViewer = ({ indexPattern, json }: RequestCodeViewerProps) => {
+  const { services } = useKibana();
+  const prepend = services.http?.basePath?.prepend;
+  const navigateToUrl = services.application?.navigateToUrl;
+  const canShowDevTools = services.application?.capabilities?.dev_tools.show;
+  const devToolsDataUri = compressToEncodedURIComponent(`GET ${indexPattern}/_search\n${json}`);
+  const devToolsUrl = `/app/dev_tools#/console?load_from=data:text/plain,${devToolsDataUri}`;
+  const shouldShowDevToolsLink = !!(indexPattern && canShowDevTools);
+
+  const handleDevToolsLinkClick = useCallback(
+    (event: MouseEvent<HTMLAnchorElement>) => {
+      event.preventDefault();
```

Comment:
There are a couple more edge cases that should be handled: 

https://github.com/elastic/kibana/blob/41f7b429d1f175f1bf1be4e3097da9d9c1c91bb4/src/plugins/kibana_react/public/app_links/click_handler.ts#L34-L44

> Since we're not using `RedirectAppLinks`, we need to use `navigateToUrl`

Why aren't we? Should we?